### DISCUSSION TITLE
Which x265 version is used?

### DISCUSSION THREAD

**Discussion by alatteri** [CESNET/UltraGrid#176] [Q&A] [answered]
Hello,

I was looking at the FFmpeg build scripts, and it is clear that x264 is being pulled and built, but I don't see the equivalent for x265.  This makes me think that the Ubuntu distribution version is being used, which generally is always a version or two behind.   Even in the latest Ubuntu beta, they are still on 3.4.2, yet the latest release of x265 is 3.5 from March.   Did I miss something in the build scripts, or would it be possible to add a fresh x265 build into the continuous integration?

Thanks,
Alan

**Comment by MartinPulec** [comment, marked as answer]
Hi, I basically agree that newer x265 version may be useful to add but I've rather thought about pulling one from some PPA, which wouldn't necessarily be the current.

Are there some benchmarks showing that eg. 3.5 is significantly better than eg. 3.3 or some new features? The point is that using a version from PPA reduces build time and perhaps it would be also better to maintain.

(Just FYI, the build system is U18.04 which has indeed an old x264. But I've found a [usable PPA](https://launchpad.net/~jonathonf/+archive/ubuntu/x265-3.3) that albeit has v3.3, but I think that there is [nothing important to lose yet](https://x265.readthedocs.io/en/release_3.5/releasenotes.html)).

**Reply by alatteri** [reply]
I was interested in this bug fix:
 `2. Encoder hang when VBV is used with slices.`

In my previous testing, enabling slices with x265 would quickly crash the encoder.  If this fixed slices, it could allow signigicant performance improvements.

**Reply by MartinPulec** [reply]
I see, thanks for the clarification. So it must be 3.5 then...

**Reply by MartinPulec** [reply]
I've added a binary version from _ppa:savoury1/vlc3_ to the continuous build - it is the v3.5. Building own version might be a bit tricky - I've tried it with default configuration and it has built only 8-bit version without any CPU features (see `x265 --version`) - I believe that linking more bit depths it a bit harder than just enabling some switches. AFAIK Debian links that together from 8+10+12 bit versions.

**Reply by alatteri** [reply]
I saw that in the commit.  Thanks, I'll try it after some of these projects wind down.  Still searching for the elusive 10bit 4K realtime encoding.
